src/App.tsx

```diff
@@ -15,33 +15,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer]
+
+  const [isVisible, setVisio] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [isSortByAlphabet, sortAlphabetically] = useState(false);
+  const [isSortByLength, sortByLength] = useState(false);
+  const [isReset, setResetedGoods] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goods.filter(good => good.length >= Number(goodsLength));
+
+  if (isSortByLength) {
+    visibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isSortByAlphabet) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+    }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  if (isReset) {
+    setReversed(false);
+    sortAlphabetically(false);
+    sortByLength(false);
+    setResetedGoods(false);
+    setGoodsLength('1');
+  }
+
+return (
+  <div className="App message">
+    {!isVisible && <div>
+      <p
+        className='title message-header'>
+          Control of goods starts from pushing this
+      </p>
+      <button
+        type="button"
+        className='button is-large is-fullwidth'
+        onClick={() =>setVisio(true)}
+      >
+        Start
     </button>
```

Comment:
```suggestion
    {!isVisible && (
      <div>
        <p
          className='title message-header'>
            Control of goods starts from pushing this
        </p>
        <button
          type="button"
          className='button is-large is-fullwidth'
          onClick={() =>setVisio(true)}
        >
          Start
      </button>
```
something went wrong

Reply:
added braces